src/wincng.c

```diff
@@ -1707,67 +1718,114 @@ _libssh2_wincng_cipher_init(_libssh2_cipher_ctx *ctx,
         return -1;
     }
 
-    if (type.dwUseIV) {
-        pbIV = malloc(dwBlockLength);
-        if (!pbIV) {
+    pbIV = NULL;
+    pbCtr = NULL;
+    dwIV = 0;
+    dwCtrLength = 0;
```

Comment:
Thanks, I like your approach to setting up these variables. I think it is more flexible than my original approach.